src/python_framework_v02/nhd_network.py

```diff
@@ -307,15 +311,15 @@ def replace_waterbodies_connections(connections, waterbodies):
     new_conn = {}
     waterbody_nets = separate_waterbodies(connections, waterbodies)
 
-    reversed_conns = reverse_network(connections)
+    #reversed_conns = reverse_network(connections)
     for n in connections:
         if n in waterbodies:
             wbody_code = waterbodies[n]
             if wbody_code in new_conn:
                 continue
 
             # get all nodes from waterbody
-            wbody_nodes = [k for k, v in waterbodies.items() if v == wbody_code]
+            wbody_nodes = nodes(waterbody_nets[wbody_code])
```

Comment:
@groutr I see we made this change here but the master reflects the original. Should we bring this over?

See [here...](https://github.com/NOAA-OWP/t-route/blob/63e1578af5edf7086df7a4106314d5f31bfdbc71/src/python_framework_v02/troute/nhd_network.py#L411)

Reply:
Maybe. I think this just narrows down the number of nodes.